pkg/db/db.go

```diff
@@ -656,3 +675,63 @@ func (g *GormDB) Insert(ctx context.Context, rec *Record) error {
 		return tx.Table(g.tableName).Create(rec).Error
 	})
 }
+
+// uid is here to fulfill the value.Context interface for the transformer.
+// This is similar to authenticatedDataString from the k8s apiserver's storage interface
+// for etcd: https://github.com/kubernetes/kubernetes/blob/a42f4f61c2c46553bfe338eefe9e81818c7360b4/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L63
+type uid string
+
+func (u uid) AuthenticatedData() []byte {
+	return []byte(u)
+}
+
+func (g *GormDB) encryptData(ctx context.Context, rec *Record) error {
+	gr := schema.GroupResource{
+		Group: rec.APIGroup,
+		// This is a hack to convert Kind to Resource. Resource is supposed to be the lowercase and plural of Kind.
+		// We will expect the provided EncryptionConfiguration to use lowercase and singular for the resource names instead, to make this easier.
+		Resource: strings.ToLower(rec.Kind),
+	}
+	logrus.Debugf("(encryptData) Finding transformer for GroupResource: %s", gr.String())
+
+	if t, exists := g.transformers[gr]; exists {
+		logrus.Debugf("Encrypting data for record %s in namespace %s", rec.Name, rec.Namespace)
+		encryptedData, err := t.TransformToStorage(ctx, []byte(rec.Data.String()), uid(rec.UID))
+		if err != nil {
+			return err
+		}
+
+		rec.Data, err = json.Marshal(map[string]string{
+			"e": base64.StdEncoding.EncodeToString(encryptedData),
+		})
+		return err
+	}
+	return nil
+}
```

Comment:
I would like particularly thorough review for this portion. The `value.Context` interface from k8s is supposed to have some kind of unique value for each item to be encrypted, I think? This is just based on reading the apiserver code that I linked. To me, it made the most sense to just use the Record's `uid` field.